src/query/sql/src/planner/plans/join.rs

```diff
@@ -423,8 +423,6 @@ impl Operator for Join {
                     | JoinType::RightSemi
                     | JoinType::RightMark
             )
-            && probe_physical_prop.distribution != Distribution::Broadcast
```

Comment:
This code was added when fixing a bug in tpcds, but now it seems to be redundant.